### PR TITLE
Add caseGroupStatus

### PR DESCRIPTION
**Pull request by jwgwalton**
- Adds status to case group, driven by state transitions from case events. 
- Audit table for Case Group Status, for traceability of changes.

### CODE REVIEW

**Comment by jwgwalton** [comment]
RE: Codacy issues

TODO is useful as the state transitions will need to be updated when other survey types come into play, EQ's etc.

File: src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseGroup.java [hunk 1]
```diff
@@ -53,4 +56,8 @@
   @Column(name = "sampleunittype")
   private String sampleUnitType;
 
+  @Enumerated(EnumType.STRING)
+  @Column(name = "casegroupstatus")
```
Comment (rjweeks) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/domain/model/CaseGroup.java]: Column name should be 'status'

File: src/main/resources/database/changes/release-10.49.0/add_case_group_status_field.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE casesvc."casegroup"
+    ADD COLUMN casegroupstatus character varying(20);
```
Comment (rjweeks) [inline src/main/resources/database/changes/release-10.49.0/add_case_group_status_field.sql]: This needs to change to just 'status'

Comment (rjweeks) [reply src/main/resources/database/changes/release-10.49.0/add_case_group_status_field.sql]: Need to update API.md to show the status in the CaseGroup record.

Comment (rjweeks) [reply src/main/resources/database/changes/release-10.49.0/add_case_group_status_field.sql]: We need to add a CaseGroupStatusAudit to record state transitions, PartyID and date/time.

File: src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseGroupAuditServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Override
+    public void updateAuditTable(final CaseGroup caseGroup, final CaseEvent caseEvent, final Case caze) {
+        CaseGroupStatusAudit auditEntity = new CaseGroupStatusAudit();
+
+        auditEntity.setCaseGroupFK(caseGroup.getCaseGroupPK());
+        auditEntity.setStatus(caseGroup.getStatus());
+        auditEntity.setPartyId(caze.getPartyId());
+        auditEntity.setCreatedDateTime(DateTimeUtil.nowUTC());
+        caseGroupStatusAuditRepository.saveAndFlush(auditEntity);
```
Comment (rjweeks) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseGroupAuditServiceImpl.java]: We need to log the audit details as well

File: src/main/resources/database/changes/release-10.50.0/add_case_group_status_audit_table.sql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+CREATE TABLE casesvc.casegroupstatusaudit (
+    casegroupstatusauditpk bigint NOT NULL,
+    casegroupfk bigint NOT NULL,
+    partyid uuid,
+    status  character varying(20),
+    createddatetime timestamp with time zone
+);
```
Comment (rjweeks) [inline src/main/resources/database/changes/release-10.50.0/add_case_group_status_audit_table.sql]: We need to create the sequence for casegroupstatusauditpk

File: src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseGroupAuditServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Override
+    public void updateAuditTable(final CaseGroup caseGroup, final CaseEvent caseEvent, final Case caze) {
+        CaseGroupStatusAudit auditEntity = new CaseGroupStatusAudit();
+
+        auditEntity.setCaseGroupFK(caseGroup.getCaseGroupPK());
+        auditEntity.setStatus(caseGroup.getStatus());
+        auditEntity.setPartyId(caze.getPartyId());
+        auditEntity.setCreatedDateTime(DateTimeUtil.nowUTC());
+        log.info("Updating the caseGroupStatus to {}, for case {} ",  caseGroup.getStatus(), caseGroup.getCaseGroupPK());
```
Comment (rjweeks) [inline src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseGroupAuditServiceImpl.java]: Need to include partyId.

File: src/test/java/uk/gov/ons/ctp/response/casesvc/state/StateTransitionManagerUnitTest.java [hunk 6]
```diff
@@ -85,4 +108,34 @@ public void testCaseTransitions() {
...
+      transitions.forEach((caseEvent, caseState) ->{
+        try {
+          CaseGroupStatus newStatus = caseGroupStageManager.transition(currentState, caseEvent);
+          assertEquals(caseState, newStatus);
+        } catch (CTPException e) {
+          fail();
+        }
+      });
+
+      Arrays.asList(CategoryDTO.CategoryName.values()).forEach(event -> {
```
Comment (benjefferies) [inline src/test/java/uk/gov/ons/ctp/response/casesvc/state/StateTransitionManagerUnitTest.java]: The loop for each transaction makes this hard to follow but I appreciate it's done for consistency
